providers/util.py

```diff
@@ -1,7 +1,15 @@
 import subprocess
+import os
 
 def call(cmdline):
     output = subprocess.check_output(cmdline)
     lines = output.splitlines()
     lines = [line.decode("utf-8") for line in lines]
     return lines
+
+def call_batctl(batadv_dev, args):
+
+    if os.geteuid() > 0:
+        return call(['sudo', 'batctl', 'meshif', batadv_dev] + args)
+
+    return call(['batctl', '-m', batadv_dev] + args)
```

Comment:
This needs refactoring and a rebase after the merge of #51 or #64.

Reply:
As soon as it does, one should write the list into a var in order not to repeat it and insert sudo at index zero before returning it.
Possibly like this:

```python3
def call_batctl(batadv_dev, args):
    base_args = ['batctl', 'meshif', batadv_dev]
    if os.geteuid() > 0:
        base_args.insert(0, 'sudo')
    return call(base_args + args)
```

Reply:
Yes, this conflicts with #51, I created #80 (replacing #51) with which the conflict should be smaller.

Reply:
I implemented it with the list both here and in #80